File: packages/jsii-rosetta/test/translations/classes/class_with_props_argument.java
public class MyClassProps {
    private String prop1;
    public String getProp1() {
        return this.prop1;
    }
    public MyClassProps prop1(String prop1) {
        this.prop1 = prop1;
        return this;
    }
    private Number prop2;
    public Number getProp2() {
        return this.prop2;
    }
    public MyClassProps prop2(Number prop2) {
        this.prop2 = prop2;
        return this;
    }
}

public class MyClass extends SomeOtherClass {
    public MyClass(Construct scope, String id, MyClassProps props) {
        super(scope, id, props);

        System.out.println(props.getProp1());
    }
}

File: packages/jsii-rosetta/test/translations/classes/class_with_props_argument.ts
interface MyClassProps {
  readonly prop1: string;
  readonly prop2: number;
}

class MyClass extends cdk.SomeOtherClass {
  constructor(scope: cdk.Construct, id: string, props: MyClassProps) {
    super(scope, id, props);

    console.log(props.prop1);
  }
}
